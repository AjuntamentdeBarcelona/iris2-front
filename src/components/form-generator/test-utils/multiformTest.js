var _ = require('lodash')

export default function multiformTest(wrapper) {
  describe('formset API', () => {
    it('should return item label', () => {
      const label = wrapper.vm.getItemLabel(wrapper.vm.value[0])
      expect(label).toBe(wrapper.vm.value[0][wrapper.vm.schema.labelField])
    })
    it('should update value', () => {
      wrapper.vm.setValue()
    })
    it('should return array of inline items', () => {
      expect(Array.isArray(wrapper.vm.getInlineItems())).toBe(true)
    })
    if (wrapper.vm.fixedSchema) {
      it('should return fixed schema for unknown items when schema is fixed', () => {
        expect(wrapper.vm.getSchemaForItem({})).toEqual(wrapper.vm.fixedSchema)
      })
    } else {
      it('should return schema for an existent item', () => {
        expect(wrapper.vm.getItemLabel(wrapper.vm.value[0])).toBeTruthy()
      })
      it('should have one form per model', () => {
        expect(wrapper.vm.forms.length).toBe(wrapper.vm.getInlineItems().length)
      })
      it('should render one form per model', () => {
        /**
         * @todo Reenable when test-util bug fixed https://github.com/MattRynbrand/test-utils-components-option-reproduction
         */
        // expect(wrapper.findAll('.iris-table-form').length).toBe(
        //  wrapper.vm.getInlineItems().length
        // )
      })
      it('should throw error when getting schema for unknown items with variable schemas', () => {
        expect(() => {
          wrapper.vm.getSchemaForItem({})
        }).toThrow()
      })
    }
    describe('Individual form event consolidation', () => {
      /**
       * @todo Reenable when test-util bug fixed https://github.com/MattRynbrand/test-utils-components-option-reproduction
       wrapper.find('input[type="text"]').setValue('Value changed')
       it('should handle item updated event', () => {
         expect(wrapper.emitted()['model-updated']).toBeTruthy()
        })
        it('should handle form validated', () => {
          expect(wrapper.emitted()['validated']).toBeTruthy()
        })
*/
    })
    describe('Error handling', () => {
      it('should process valid error array', () => {
        const error = 'This is a test error'
        const result = wrapper.vm.processFormErrors([
          {
            1: {
              test: [error],
            },
          },
        ])
        expect(Array.isArray(result[1])).toBe(true)
        expect(result[1].length).toBe(1)
        expect(result[1][0].error).toBe(error)
      })
      it('should return empty errors', () => {
        expect(wrapper.vm.processFormErrors()).toEqual({})
      })
    })
  })
}
