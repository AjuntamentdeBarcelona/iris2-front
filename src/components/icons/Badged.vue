
<!-- 
<template>
  <svg
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    x="0px"
    y="0px"
    viewBox="0 0 389.9 483.3"
    style="enable-background:new 0 0 389.9 483.3;"
    xml:space="preserve">
    <g id="escutbcn">
      <path
        d="M216.8,75.6c-2.5,0.8-4.3,3.2-5.1,6.3c2.4,2,5.2,2.9,7.8,2.1c2.5-0.8,4.3-3.2,5-6.2l0,0c-0.5,3.1,0.1,6,2.1,7.8
          s4.9,2.1,7.9,1.3c0.6-3.2-0.1-6-2.1-7.9c-2-1.8-4.9-2.1-7.9-1.2c3-1.1,5.1-3.1,5.7-5.7c0.6-2.6-0.6-5.3-2.9-7.5
          c-3,1.1-5.2,3.1-5.7,5.7c-0.5,2.7,0.6,5.3,2.9,7.6C222.2,75.6,219.3,74.7,216.8,75.6 M259.8,82.3c1.1-2.5,0.5-5.3-1.3-8
          c-3.1,0.5-5.7,2-6.8,4.4s-0.5,5.3,1.3,8C256.2,86.3,258.7,84.8,259.8,82.3 M253,86.7c-1.9-2.5-4.6-4-7.2-3.7
          c-2.6,0.3-4.8,2.2-6.3,5.1c2,2.5,4.5,3.9,7.2,3.7C249.4,91.5,251.6,89.6,253,86.7 M195,454.2L28.4,287.7L195,121.1l166.6,166.5
          L195,454.2z M286.8,194.1c-57.9-57.8-91.5-76.6-91.6-76.6l-0.2-0.1l-0.3,0.1c0,0-34.6,19.8-93.3,78.3
          c-57.9,57.9-76.7,91.5-76.7,91.5l-0.2,0.3l0.2,0.2c0,0,19.8,34.7,78.4,93.3c57.9,57.8,91.5,76.6,91.5,76.6l0.3,0.2l0.3-0.2
          c0,0,34.7-19.8,93.2-78.3c57.9-57.9,76.7-91.5,76.7-91.6l0.1-0.2l-0.1-0.2C365.2,287.4,345.4,252.8,286.8,194.1 M9.8,280.8
          c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9s6.9-3.1,6.9-6.9S13.6,280.8,9.8,280.8 M195,466c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9
          c3.8,0,7-3.1,7-6.9C201.9,469.1,198.8,466,195,466 M380.2,280.8c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9s6.9-3.1,6.9-6.9
          S384,280.8,380.2,280.8 M133.4,64.8c-2.8-2.3-4.5-5.3-4.5-8.6c0-7.1,7.6-12.8,17.1-12.8c2.3,0,4.4,0.3,6.3,0.9
          c-0.3,1.4-0.5,2.9-0.5,4.4c0,4.7,1.6,9,4.4,12.7l7.7-7.7c-0.9-1.5-1.4-3.2-1.4-5c0-7.1,7.6-12.8,17-12.8c3.8,0,7.4,0.9,10.2,2.6
          v15.2h10.7V38.2c2.8-1.5,6.1-2.3,9.8-2.3c9.4,0,17,5.7,17,12.8c0,1.8-0.5,3.5-1.4,5l7.7,7.7c2.8-3.7,4.4-8,4.4-12.7
          c0-1.5-0.2-3-0.5-4.4c1.9-0.6,4.1-0.9,6.4-0.9c9.4,0,17.1,5.7,17.1,12.8c0,3.3-1.7,6.3-4.5,8.6l7.5,7.5c4.7-4.2,7.6-9.9,7.6-16.1
          c0-13-12.4-23.5-27.7-23.5c-4,0-7.8,0.7-11.3,2c-5-5.8-13.2-9.6-22.3-9.6c-5.7,0-11,1.5-15.3,3.9c-4.4-2.5-9.7-3.9-15.3-3.9
          c-9.2,0-17.3,3.8-22.3,9.6c-3.4-1.3-7.2-2-11.3-2c-15.3,0-27.7,10.5-27.7,23.5c0,6.2,2.9,11.9,7.5,16.1L133.4,64.8z M142.9,91.8
          c2.7,0.2,5.2-1.1,7.2-3.7c-1.4-2.9-3.6-4.8-6.3-5.1c-2.6-0.3-5.2,1.2-7.1,3.7C138.1,89.6,140.3,91.5,142.9,91.8 M187.1,74.1
          c-2.3,1.4-3.6,4-3.7,7.2c2.8,1.5,5.8,1.8,8.1,0.4c2.3-1.3,3.6-4,3.6-7.2l0,0c0.1,3.2,1.4,5.9,3.7,7.2c2.3,1.3,5.3,1.1,8.1-0.4
          c-0.1-3.2-1.3-5.8-3.7-7.2c-2.3-1.3-5.2-1.1-8,0.4c2.7-1.7,4.4-4.1,4.4-6.8c0-2.7-1.7-5.1-4.4-6.7c-2.7,1.7-4.4,4.1-4.4,6.7
          c0,2.7,1.7,5.1,4.4,6.8C192.3,73,189.4,72.8,187.1,74.1 M195,17.9c4,0,7.2-3.2,7.2-7.2S199,3.5,195,3.5s-7.3,3.2-7.3,7.2
          C187.8,14.7,191,17.9,195,17.9 M138,78.8c-1.1-2.5-3.6-4-6.8-4.4c-1.8,2.6-2.4,5.5-1.3,8c1.1,2.4,3.7,4,6.8,4.4
          C138.4,84.1,139.1,81.2,138,78.8 M157,78.9c-2,1.8-2.6,4.7-2.1,7.9c3,0.9,6,0.5,8-1.3s2.6-4.7,2-7.8c0.8,3.1,2.6,5.5,5.1,6.2
          c2.5,0.9,5.4-0.1,7.7-2.1c-0.7-3.1-2.5-5.5-5.1-6.3c-2.5-0.8-5.3,0-7.7,2.1c2.3-2.2,3.4-4.9,2.8-7.5c-0.5-2.6-2.7-4.6-5.7-5.7
          c-2.3,2.2-3.4,4.9-2.9,7.5s2.7,4.6,5.7,5.7C162,76.8,159.1,77.1,157,78.9 M195,99.1c16.9,0,32.7,4.1,46.2,10.3l5.9-9
          c-15.2-7.2-33-11.6-52-11.6s-36.8,4.2-52.2,11.6l6,9C162.3,103.3,178.3,99.1,195,99.1 M197.4,290.2h34v42.2h-34V290.2z
          M192.6,285.3h-33.9v-42.1h33.9V285.3z M192.6,226.2h-33.9v-58l-17,16.9v41.1h-41l-17,17h58v42.1H41.5l-2.4,2.4l2.4,2.4h15.3v15.4
          l16.9,16.9v-32.3h17v49.2l17,17v-66.2h17v83.2l17,16.9v-100h17v117l17,17v-134h16.9v150.9l2.4,2.4l2.4-2.4v-91.8h34v58l16.9-17v-41
          h41l16.9-16.9h-57.9v-42.2h100.1l2.5-2.5l-2.4-2.4h-15.3V270l-17-17v32.3h-16.9V236l-17-16.9v66.2h-17v-83.1l-16.9-17v100.1h-16.9
          v-117l-17-16.9v134h-17V134.3l-2.4-2.4l-2.5,2.5L192.6,226.2L192.6,226.2z"/>
      <g id="creu">
        <polygon
          class="creu-container"
          points="192.6,134.2 192.6,285.3 41.5,285.3 	"/>
        <polygon
          class="creu-container"
          points="197.5,441.3 197.5,290.2 348.6,290.2 	"/>
        <polygon
          class="creu"
          points="158.7,285.3 158.7,243.2 192.6,243.2 192.6,226.2 192.6,226.2 158.7,226.2 158.7,168.2 141.7,185.1
          141.7,226.2 100.7,226.2 83.7,243.2 141.7,243.2 141.7,285.3 	"/>
        <polygon
          class="creu"
          points="231.4,290.2 231.4,332.3 197.5,332.3 197.5,349.3 197.5,349.3 231.4,349.3 231.4,407.3 248.4,390.4
          248.4,349.3 289.4,349.3 306.4,332.3 248.4,332.3 248.4,290.2 	"/>
      </g>
    </g>
  </svg>
</template>

<script>
export default {
  name: 'Badged',
  components: {},
}
</script>

<style lang="scss">
#escutbcn {
  fill-opacity: 1;
  fill-rule: nonzero;
  stroke: none;
}
.creu {
  fill: $red;
  &-container {
    fill: $white;
  }
}
</style>

-->

<!-- LOGO CORUÃ‘A -->


<template>

  <svg
    xmlns="http://www.w3.org/2000/svg" 
    viewBox="0 0 20.73 30.91" 
    class="logo-escut">
    <defs>
      <style>.cls-11{fill:#fff;}</style>
    </defs>
    <g 
      id="Capa_2" 
      data-name="Capa 2">
      <g 
        id="Capa_1-2" 
        data-name="Capa 1">
        <path 
          class="cls-11" 
          d="M9.12.26a11,11,0,0,1,1.57.57l1.82.83a8.58,8.58,0,0,0,.94.37,2.66,2.66,0,0,0,.83.13,2,2,0,0,0,1.59-.65A5.22,5.22,0,0,0,16.68,0h1.9a9.92,9.92,0,0,1-1.72,3.93,4.33,4.33,0,0,1-3.68,1.69,5.72,5.72,0,0,1-1.35-.16A8.44,8.44,0,0,1,10.49,5L9.18,4.45C8,4,7.32,3.67,7.08,3.58a2.13,2.13,0,0,0-.76-.13,2,2,0,0,0-1.69.82A4.09,4.09,0,0,0,4,5.62H2.16A10.28,10.28,0,0,1,3.37,2.35,4.66,4.66,0,0,1,7.53,0,5.72,5.72,0,0,1,9.12.26"/>
        <path 
          class="cls-11" 
          d="M0,19.72v-.06A11.11,11.11,0,0,1,11.4,8.42a10.62,10.62,0,0,1,9.18,4.51l-4.51,3.49a5.79,5.79,0,0,0-4.73-2.54c-3,0-5.16,2.57-5.16,5.72v.06c0,3.25,2.13,5.79,5.16,5.79,2.25,0,3.58-1.06,4.88-2.63L20.73,26a10.91,10.91,0,0,1-9.58,4.89A11,11,0,0,1,0,19.72"/>
      </g>
    </g>
  </svg>

</template>

<script>
export default {
  name: 'Badged',
  components: {},
}
</script>
