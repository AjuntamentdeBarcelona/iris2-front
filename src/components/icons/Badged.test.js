import { mount } from '@vue/test-utils'
import Badged from './Badged.vue'

describe('Badged component', () => {
  const wrapper = mount(Badged)

  it('is a Vue instance', () => {
    const actual = wrapper.html()
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(actual).toMatchSnapshot()
  })
})
