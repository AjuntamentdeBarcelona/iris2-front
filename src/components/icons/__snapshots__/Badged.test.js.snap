// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badged component is a Vue instance 1`] = `
"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 389.9 483.3\\" xml:space=\\"preserve\\" style=\\"enable-background: new 0 0 389.9 483.3;\\"><g id=\\"escutbcn\\"><path d=\\"M216.8,75.6c-2.5,0.8-4.3,3.2-5.1,6.3c2.4,2,5.2,2.9,7.8,2.1c2.5-0.8,4.3-3.2,5-6.2l0,0c-0.5,3.1,0.1,6,2.1,7.8
        s4.9,2.1,7.9,1.3c0.6-3.2-0.1-6-2.1-7.9c-2-1.8-4.9-2.1-7.9-1.2c3-1.1,5.1-3.1,5.7-5.7c0.6-2.6-0.6-5.3-2.9-7.5
        c-3,1.1-5.2,3.1-5.7,5.7c-0.5,2.7,0.6,5.3,2.9,7.6C222.2,75.6,219.3,74.7,216.8,75.6 M259.8,82.3c1.1-2.5,0.5-5.3-1.3-8
        c-3.1,0.5-5.7,2-6.8,4.4s-0.5,5.3,1.3,8C256.2,86.3,258.7,84.8,259.8,82.3 M253,86.7c-1.9-2.5-4.6-4-7.2-3.7
        c-2.6,0.3-4.8,2.2-6.3,5.1c2,2.5,4.5,3.9,7.2,3.7C249.4,91.5,251.6,89.6,253,86.7 M195,454.2L28.4,287.7L195,121.1l166.6,166.5
        L195,454.2z M286.8,194.1c-57.9-57.8-91.5-76.6-91.6-76.6l-0.2-0.1l-0.3,0.1c0,0-34.6,19.8-93.3,78.3
        c-57.9,57.9-76.7,91.5-76.7,91.5l-0.2,0.3l0.2,0.2c0,0,19.8,34.7,78.4,93.3c57.9,57.8,91.5,76.6,91.5,76.6l0.3,0.2l0.3-0.2
        c0,0,34.7-19.8,93.2-78.3c57.9-57.9,76.7-91.5,76.7-91.6l0.1-0.2l-0.1-0.2C365.2,287.4,345.4,252.8,286.8,194.1 M9.8,280.8
        c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9s6.9-3.1,6.9-6.9S13.6,280.8,9.8,280.8 M195,466c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9
        c3.8,0,7-3.1,7-6.9C201.9,469.1,198.8,466,195,466 M380.2,280.8c-3.8,0-6.9,3.1-6.9,6.9s3.1,6.9,6.9,6.9s6.9-3.1,6.9-6.9
        S384,280.8,380.2,280.8 M133.4,64.8c-2.8-2.3-4.5-5.3-4.5-8.6c0-7.1,7.6-12.8,17.1-12.8c2.3,0,4.4,0.3,6.3,0.9
        c-0.3,1.4-0.5,2.9-0.5,4.4c0,4.7,1.6,9,4.4,12.7l7.7-7.7c-0.9-1.5-1.4-3.2-1.4-5c0-7.1,7.6-12.8,17-12.8c3.8,0,7.4,0.9,10.2,2.6
        v15.2h10.7V38.2c2.8-1.5,6.1-2.3,9.8-2.3c9.4,0,17,5.7,17,12.8c0,1.8-0.5,3.5-1.4,5l7.7,7.7c2.8-3.7,4.4-8,4.4-12.7
        c0-1.5-0.2-3-0.5-4.4c1.9-0.6,4.1-0.9,6.4-0.9c9.4,0,17.1,5.7,17.1,12.8c0,3.3-1.7,6.3-4.5,8.6l7.5,7.5c4.7-4.2,7.6-9.9,7.6-16.1
        c0-13-12.4-23.5-27.7-23.5c-4,0-7.8,0.7-11.3,2c-5-5.8-13.2-9.6-22.3-9.6c-5.7,0-11,1.5-15.3,3.9c-4.4-2.5-9.7-3.9-15.3-3.9
        c-9.2,0-17.3,3.8-22.3,9.6c-3.4-1.3-7.2-2-11.3-2c-15.3,0-27.7,10.5-27.7,23.5c0,6.2,2.9,11.9,7.5,16.1L133.4,64.8z M142.9,91.8
        c2.7,0.2,5.2-1.1,7.2-3.7c-1.4-2.9-3.6-4.8-6.3-5.1c-2.6-0.3-5.2,1.2-7.1,3.7C138.1,89.6,140.3,91.5,142.9,91.8 M187.1,74.1
        c-2.3,1.4-3.6,4-3.7,7.2c2.8,1.5,5.8,1.8,8.1,0.4c2.3-1.3,3.6-4,3.6-7.2l0,0c0.1,3.2,1.4,5.9,3.7,7.2c2.3,1.3,5.3,1.1,8.1-0.4
        c-0.1-3.2-1.3-5.8-3.7-7.2c-2.3-1.3-5.2-1.1-8,0.4c2.7-1.7,4.4-4.1,4.4-6.8c0-2.7-1.7-5.1-4.4-6.7c-2.7,1.7-4.4,4.1-4.4,6.7
        c0,2.7,1.7,5.1,4.4,6.8C192.3,73,189.4,72.8,187.1,74.1 M195,17.9c4,0,7.2-3.2,7.2-7.2S199,3.5,195,3.5s-7.3,3.2-7.3,7.2
        C187.8,14.7,191,17.9,195,17.9 M138,78.8c-1.1-2.5-3.6-4-6.8-4.4c-1.8,2.6-2.4,5.5-1.3,8c1.1,2.4,3.7,4,6.8,4.4
        C138.4,84.1,139.1,81.2,138,78.8 M157,78.9c-2,1.8-2.6,4.7-2.1,7.9c3,0.9,6,0.5,8-1.3s2.6-4.7,2-7.8c0.8,3.1,2.6,5.5,5.1,6.2
        c2.5,0.9,5.4-0.1,7.7-2.1c-0.7-3.1-2.5-5.5-5.1-6.3c-2.5-0.8-5.3,0-7.7,2.1c2.3-2.2,3.4-4.9,2.8-7.5c-0.5-2.6-2.7-4.6-5.7-5.7
        c-2.3,2.2-3.4,4.9-2.9,7.5s2.7,4.6,5.7,5.7C162,76.8,159.1,77.1,157,78.9 M195,99.1c16.9,0,32.7,4.1,46.2,10.3l5.9-9
        c-15.2-7.2-33-11.6-52-11.6s-36.8,4.2-52.2,11.6l6,9C162.3,103.3,178.3,99.1,195,99.1 M197.4,290.2h34v42.2h-34V290.2z
        M192.6,285.3h-33.9v-42.1h33.9V285.3z M192.6,226.2h-33.9v-58l-17,16.9v41.1h-41l-17,17h58v42.1H41.5l-2.4,2.4l2.4,2.4h15.3v15.4
        l16.9,16.9v-32.3h17v49.2l17,17v-66.2h17v83.2l17,16.9v-100h17v117l17,17v-134h16.9v150.9l2.4,2.4l2.4-2.4v-91.8h34v58l16.9-17v-41
        h41l16.9-16.9h-57.9v-42.2h100.1l2.5-2.5l-2.4-2.4h-15.3V270l-17-17v32.3h-16.9V236l-17-16.9v66.2h-17v-83.1l-16.9-17v100.1h-16.9
        v-117l-17-16.9v134h-17V134.3l-2.4-2.4l-2.5,2.5L192.6,226.2L192.6,226.2z\\"></path> <g id=\\"creu\\"><polygon points=\\"192.6,134.2 192.6,285.3 41.5,285.3 	\\" class=\\"creu-container\\"></polygon> <polygon points=\\"197.5,441.3 197.5,290.2 348.6,290.2 	\\" class=\\"creu-container\\"></polygon> <polygon points=\\"158.7,285.3 158.7,243.2 192.6,243.2 192.6,226.2 192.6,226.2 158.7,226.2 158.7,168.2 141.7,185.1
        141.7,226.2 100.7,226.2 83.7,243.2 141.7,243.2 141.7,285.3 	\\" class=\\"creu\\"></polygon> <polygon points=\\"231.4,290.2 231.4,332.3 197.5,332.3 197.5,349.3 197.5,349.3 231.4,349.3 231.4,407.3 248.4,390.4
        248.4,349.3 289.4,349.3 306.4,332.3 248.4,332.3 248.4,290.2 	\\" class=\\"creu\\"></polygon></g></g></svg>"
`;
